src/command.rs

```diff
@@ -93,9 +93,13 @@ pub async fn add(registry: RegistryUri, dependency: &str) -> eyre::Result<()> {
 
     let mut manifest = Manifest::read().await?;
 
-    manifest
-        .dependencies
-        .push(Dependency::new(registry, repository, package, version));
+    manifest.dependencies.push(Dependency::new(
+        registry,
+        repository,
+        package,
+        version,
+        crate::manifest::DependencyType::Root,
```

Comment:
root is not serialized i guess?
why is it needed?

Reply:
It's not needed here, it's used to build the dependency graph while allowing manifest-derived dependencies to override the lockfile metadata, but I haven't thought about serialisation so I'll confirm if that works sensibly.

Reply:
Ok, just checked and yes it gets ignored as it's supposed to.

Reply:
Removed it after all. 🤷